### PR TITLE
Move expiration check to ACLGenerator

### LINKED ISSUE #33: Hoist "expiration" impl into core
Currently the implementation of the 'expiration' property is handled through a block of copy-paste code in every generator. Should a generator author forget to paste in this code, expiration won't work.

The core application should handle the expiration implementation and not even pass it through to the generator.

### PR DESCRIPTION
**Pull request by btriller**
fixes #33

### CODE REVIEW

File: aerleon/lib/aclgenerator.py [hunk 1]
```diff
@@ -302,7 +304,34 @@ def __init__(self, pol, exp_info):
         self.policy = pol
         all_err = []
         all_warn = []
+        current_date = datetime.datetime.utcnow().date()
+        exp_info_date = current_date + datetime.timedelta(weeks=exp_info)
         for header, terms in pol.filters:
+            filter_name = header.FilterName(self._PLATFORM)
+            terms[:] = [
```
Comment (ankenyr) [inline aerleon/lib/aclgenerator.py:311]: What made you choose to do a list comprehension in this instance? Once list comprehensions get past a one liner they tend to become harder to read.

Comment (btriller) [reply aerleon/lib/aclgenerator.py:311]: Started out as a oneliner and grew after adding logging. Should I change to `filter()` then?

Comment (ankenyr) [reply aerleon/lib/aclgenerator.py]: They always do haha
Yea a filter would be more readable here.

**Review by ankenyr — APPROVED** [review]

**Comment by ankenyr** [comment]
@all-contributors lets add @btriller for code

**Pull request merged**